firestore/options.go

```diff
@@ -54,9 +54,13 @@ func (e exists) String() string {
 }
 
 // LastUpdateTime returns a Precondition that checks that a resource must exist and
-// must have last been updated at the given time. If the check fails, the write
-// does not occur.
-func LastUpdateTime(t time.Time) Precondition { return lastUpdateTime(t) }
+// must have last been updated at the given time, rounded to microsecond precision.
+// If the check fails, the write does not occur.
```

Comment:
I'm a little confused how this bug got noticed. The example [here](https://pkg.go.dev/cloud.google.com/go/firestore#hdr-Preconditions) shows pulling the `UpdateTime` from a document snapshot and then using it immediately in a precondition. If this workflow is followed, I don't think `UpdateTime` would ever have more than microsecond precision? Is there some other way you would calculate an `UpdateTime`?

In any case, this seems fine; it might be worth checking for additional incongruities between `time.Time` and firestore time.

Reply:
In order to see this, you have to do some slightly strange things. Any time returned on a document for update already is microsecond aligned. I also couldn't make it happen with time.Now().

However, if I create a time from nanoseconds and ensure it isn't aligned, that will result in the failure.